### ISSUE TITLE
Need user documentation, tutorials

### ISSUE THREAD

**Issue by JimMarlowe** [AtomicGameEngine/AtomicGameEngine#1461]
The wiki is a fine start, but there needs to be a bootcamp or something to get users bridged from bringing up the editor to then running an example to see how things work.

**Comment by Alan-FGR** [comment]
I agree, everybody is complaining about the lack of tutorials. I've even been PMed about that. Everywhere I look there are people having a hard time getting started: Gitter, Forums, Twitter, Reddit, and obviously here on GitHub.

I propose doing written tutorials collaboratively (so we all can review and improve them), and then somebody with a nice soothing voice and good pronunciation record a video following the steps on the tutorial. I think that's the best way to achieve decent quality contents quickly, because if each one of us publish our own independent tutorials in our blogs or something, chances are it's not gonna be that good, and we'll end up with either videos or written tutorials with no correlation.

What do you guys have in mind to tackle that problem?

Also, if we're gonna do something like that please let's not use the GitHub Wiki because it sucks that much. Let's just make a repo so we can use a sane file format or some external collaboration system.

**Comment by MuffinManKen** [comment]
IMHO, don't worry about videos.  Tutorials and documentation are desperately needed and the time required to make videos will just slow things down.  Some well-written tutorials would give existing You Tubers some material to work with.

**Comment by Alan-FGR** [comment]
I agree with @MuffinManKen. I vote for collaboratively writing and reviewing some tutorials asap so people can at least start using the engine, and later we can use them to make the videos if necessary. The problem is the Editor isn't quite ready for prime time yet, and possibly will change considerably, but we could make tutorials on how to use Atomic as a framework. We need a to-do list though. Another thing I think we should have is a Unity->Atomic API conversion table. It's tedious work though, maybe we could go after funding for that because ain't nobody willing to do that voluntarily :P .

**Comment by MuffinManKen** [comment]
Documents explaining how to transition from other popular engines (Unity, Cocos2D, etc) would be nice to have, but I think it would be pretty far down the list in terms of value.  

I've been a game developer for a fair while now, and when I look at the "docs" available for Atomic, I'm lost as to where to even start.

**Comment by Alan-FGR** [comment]
@MuffinManKen I agree, I was suggesting getting funding for the whole shebang, not just the conversion table which sure isn't priority.
I've been developing games since I couldn't ride in the front seat and since the API is really sensible and frankly, just beautiful, I could pick it up quickly, in a day I was familiar already. With a bit of help from the guys @ Gitter I could also perform things like manual drawing in a day too, all in C#; at that point Atomic far surpassed in terms of deliberate limitations what Unity offers for example, so it was a pleasant experience. I even went further and played a bit with the native code to add features and expose more functions, until I broke the bindings generator badly :P.
I have a couple of C# tutorials in the works, including one showing how to modify the native code when you need, what is quite simple but unfortunately I was having some issues and at the time nobody could help me on Gitter (it's not trivial) and these days I really don't have the time unfortunately.
Provided the bindings generator won't let you down, Atomic provides a managed development environment far less limited than any other engine out there, and if setup correctly it allows you to very easily modify the native code (which is clean and very well designed) to add performance critical features like grid pathfinding or a verlet physics engine, while still allowing you to code primarily in the managed language for productivity. This is something no other engine offers so far, maybe it will be a thing in CryEngine because although I don't know how the bindings work, I've seem some SWIG stuff in the source, so possibly you can interop quite easily too. But I don't think it's a direct competitor anyway.

**Comment by JoshEngebretson** [comment]
Quoting @Alan-FGR, "Atomic provides a managed development environment far less limited than any other engine out there, and if setup correctly it allows you to very easily modify the native code (which is clean and very well designed) to add performance critical features" 👍 

I agree, and can also see that it is difficult to initially grok how it all hangs together.  This is an area which needs improvement.  Also, Atomic can be driven in JavaScript/TypeScript/C#/C++, or any combination of these.  This is a range of developers and environments which need good getting started documentation.

@MuffinManKen Which languages are you most interested in with Atomic, for projects JavaScript/TypeScript/C# have seen the most attention, C++ is a camel of a different color.  I spun up a thread and branch which shows how setup a native C++ module, with automated scripting support, and am looking for help on improving the C++ process: https://discourse.atomicgameengine.com/t/improving-atomic-c-project-support/312

**Comment by MuffinManKen** [comment]
@Alan-FGR I guess I just need to clear some time and dive in. :) 

You guys don't need to sell me on the features. I started looking into Atomic because it was one of the few engines that ticked pretty much all the boxes on my wishlist.

@JoshEngebretson My initial plan is to use Typescript and C++. The exact balance between them, I won't really know until I dig into the engine more deeply.

Assuming I do go ahead with using Atomic on my next project, I would definitely try to contribute back in some way.  

The C++ API reference looks like Doxygen.  How is the Typescript reference generated? Because it has zero information about any of the classes/methods other then their definition.

**Comment by JoshEngebretson** [comment]
@MuffinManKen The TypeScript documentation is generated from the ambient TypeScript files, which are in themselves pretty solid API reference, for examples check the Script/TypeScript folder after a build.  There is a doc task, which uses typedoc: https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Build/Scripts/BuildCommon.js#L206

The generated docs have the class signatures, in addition methods should have any native method documentation included.  What I do note is that properties (which are generated from getter/setter methods) don't have those docs, and that is an easy area for improvement: https://github.com/AtomicGameEngine/AtomicGameEngine/issues/1552
